numba/tests/test_parfors.py

```diff
@@ -1850,6 +1850,37 @@ def test_impl(x):
         x = np.ones((2, 2, 2, 2, 2, 15))
         self.check(test_impl, x)
 
+    def test_tuple_arg(self):
+        def test_impl(x, sz):
+            for i in numba.pndindex(sz):
+                x[i] = 1
+            return x
+        sz = (10,10)
+        self.check(test_impl, np.empty(sz), sz)
+
+    def test_tuple_arg_literal(self):
+        def test_impl(x, sz):
+            for i in numba.pndindex(sz):
+                x[i] = 1
+            return x
+        sz = (10,numba.literally(10))
```

Comment:
I'm not sure that this has the desired effect. `numba.literally` outside of compiled code is a pass through NOP:
https://github.com/numba/numba/blob/4a3d575f20c1f2ef7d4a661a6f52102b4039a40e/numba/misc/special.py#L64-L87

it might be necessary to close over or locally define a const to get a literal if that's the intent?

Reply:
I've added some tests and tried to generate the right tuples using what I think you were suggesting above with one literal arg but there is some lowering error I don't understand.  Can you take a look at it?